src/transcript-plugin.tsx

```diff
@@ -1,458 +1,306 @@
   private _isLoading = false;
   private _hasError = false;
-  private _entryId = "";
-  private _captionsList: KalturaCaptionAsset[] = []; // list of captions
-  private _captions: CaptionItem[] = []; // parsed captions
-  private _kalturaClient = new KalturaClient();
-  private _transcriptLanguage = "default";
-  private _triggeredByKeyboard  = false;
-
-  constructor(
-      private _contribServices: ContribServices,
-      private _configs: ContribPluginConfigs<TranscriptPluginConfig>,
-      private _player: KalturaPlayerTypes.Player
-  ) {}
+  private _triggeredByKeyboard = false;
 
-  onPluginSetup(): void {
-    const { playerConfig } = this._configs;
+  private _removePopoverIcon: null | Function = null;
 
-    this._kalturaClient.setOptions({
-      clientTag: "playkit-js-transcript",
-      endpointUrl: playerConfig.provider.env.serviceUrl
-    });
+  private _transcriptPanel = null;
+  private _pluginState: PluginStates | null = null;
 
-    this._kalturaClient.setDefaultRequestOptions({
-        ks: playerConfig.provider.ks ? playerConfig.provider.ks : playerConfig.session.ks
-    });
-  
-    this._player.addEventListener(this._player.Event.TIME_UPDATE, this._onTimeUpdate);
-    this._player.addEventListener(this._player.Event.TEXT_TRACK_CHANGED, this._loadCaptions);
-    this._player.addEventListener(this._player.Event.RESIZE, this._updateKitchenSink);
+  constructor(name: string, player: KalturaPlayerTypes.Player, config: TranscriptConfig) {
+    super(name, player, config);
   }
 
-  onRegisterUI(): void {}
-
-  onMediaLoad(): void {
-    const { playerConfig } = this._configs;
-    this._entryId = playerConfig.sources.id;
-    this._getCaptionsList();
+  get sidePanelsManager() {
+    return this.player.getService('sidePanelsManager') as any;
   }
 
-  onMediaUnload(): void {
-    this._reset();
+  get cuePointManager() {
+    return this.player.getService('kalturaCuepoints') as any;
   }
 
-  private _reset(): void {
-    if (this._kitchenSinkItem) {
-      this._contribServices.kitchenSinkManager.remove(this._kitchenSinkItem);
-      this._kitchenSinkItem = null;
-    }
-    if (this._upperBarItem) {
-      this._contribServices.upperBarManager.remove(this._upperBarItem);
-      this._upperBarItem = null;
-    }
-    this._captionsList = [];
-    this._captions = [];
-    this._isLoading = false;
-    this._hasError = false;
-    this._entryId = "";
-    this._transcriptLanguage = "default";
+  private get _data() {
+    return this._captionMap.get(this._activeCaptionMapId) || [];
   }
 
-  private _initKitchensinkAndUpperBarItems(): void {
-    if (!this._upperBarItem && !this._kitchenSinkItem) {
-      this._addPopoverIcon();
-      this._addKitchenSinkItem();
+  loadMedia(): void {
+    if (!this.cuePointManager || !this.sidePanelsManager) {
+      this.logger.warn("kalturaCuepoints or sidePanelsManager haven't registered");
+      return;
     }
+    this.onPluginSetup();
+    this._registerCuePointTypes();
```

Comment:
Transcript plugin interested only on `this.cuePointManager.CuepointType.CAPTION` type. 
Do here:
`this.cuePointManager.registerTypes([this.cuePointManager.CuepointType.CAPTION]);`
and remove `_registerCuePointTypes` method;